src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -606,9 +601,35 @@ bool AletheProofPostprocessCallback::update(Node res,
     case ProofRule::THEORY_REWRITE:
     {
       ProofRewriteRule di;
-      rewriter::getRewriteRule(args[0], di);
-      return updateTheoryRewriteProofRewriteRule(
-          res, children, args, cdp, di);
+      bool success = true;
+
+      if (rewriter::getRewriteRule(args[0], di))
+      {
+        std::stringstream ss;
+        ss << "\"" << di << "\"";
+        Node rule = NodeManager::mkRawSymbol(ss.str(), nm->sExprType());
+
+        if (!updateTheoryRewriteProofRewriteRule(res, children, args, cdp, di))
+        {
+          success = false;
+          new_args.push_back(rule);
+        }
+      }
+      else
+      {
+        success = false;
+      }
+
+      if (!success)
+      {
+        return addAletheStep(AletheRule::HOLE,
+                             res,
+                             nm->mkNode(Kind::SEXPR, d_cl, res),
+                             children,
+                             new_args,
+                             *cdp);
+      }
+      return true;
```

Comment:
```suggestion
      if (rewriter::getRewriteRule(args[0], di))
      {
        if (updateTheoryRewriteProofRewriteRule(res, children, args, cdp, di))
        {
          return true;
        }
        std::stringstream ss;
        ss << "\"" << di << "\"";
        new_args.push_back(NodeManager::mkRawSymbol(ss.str(), nm->sExprType()));
      }
      return addAletheStep(AletheRule::HOLE,
                           res,
                           nm->mkNode(Kind::SEXPR, d_cl, res),
                           children,
                           new_args,
                           *cdp);
```

The control flow was hard to follow, so I modified it a bit.